hcl2template/types.variables.go

```diff
@@ -246,6 +249,50 @@ var variableBlockSchema = &hcl.BodySchema{
 	},
 }
 
+func decodeLocalBlock(locals []*LocalBlock, block *hcl.Block) (*LocalBlock, hcl.Diagnostics) {
+	for _, local := range locals {
+		if local.Name == block.Labels[0] {
+			return nil, []*hcl.Diagnostic{{
+				Severity: hcl.DiagError,
+				Summary:  "Duplicate variable",
+				Detail:   "Duplicate " + block.Labels[0] + " variable definition found.",
+				Context:  block.DefRange.Ptr(),
+			}}
+		}
```

Comment:
I think this won't detect duplicate variables that are defined in another file. I'd maybe add a test to verify this.

Reply:
As for a suggestion, I'd maybe change the code to be a bit similar to the one for variables:
With the following call : `moreDiags := c.LocalVariables.decodeLocalBlock(block, ectx)` similarly to what in `decodeInputVariables`

Reply:
I had the same question in my head but then I realized is checking every existing block of `locals []*LocalBlock`. I tested locally and it catches duplication. Adding a test would be nice to make it more visible.

Reply:
Unless is like in another file 🤔

Reply:
yes, `parseLocalVariables(f *hcl.File)` will be called on each hcl file.

Reply:
( FYI the current code does not look for value duplicate among the HCL locals map because HCL does that for us )

Reply:
Ok you're right, this doesn't catch duplicates in separate files. Reworking.